runtime/engine.cpp

```diff
@@ -401,17 +448,35 @@ bool api::Engine::eval_toplevel(const char *path, MutableHandleValue result) {
   // the shrinking GC causes them to be intermingled with other objects. I.e.,
   // writes become more fragmented due to the shrinking GC.
   // https://github.com/fastly/js-compute-runtime/issues/224
-  JS::PrepareForFullGC(cx);
-  JS::NonIncrementalGC(cx, JS::GCOptions::Normal, JS::GCReason::API);
+  if (isWizening()) {
+    JS::PrepareForFullGC(cx);
+    JS::NonIncrementalGC(cx, JS::GCOptions::Normal, JS::GCReason::API);
+  }
 
   // Ignore the first GC, but then print all others, because ideally GCs
   // should be rare, and developers should know about them.
   // TODO: consider exposing a way to parameterize this, and/or specifying a
   // dedicated log target for telemetry messages like this.
   JS_SetGCCallback(cx, gc_callback, nullptr);
 
+  TOPLEVEL_EVALUATED = true;
```

Comment:
Strictly speaking in the case of TLA, the top-level hasn't been fully evaluated at this point. Perhaps we can call this something else like `execution_started`?

Reply:
I filed #64 to track improving this more generally